scan-command/src/main/java/io/ballerina/scan/utils/ScanUtils.java

```diff
@@ -276,6 +301,242 @@ private static void unzipReportResources(InputStream source, File target) throws
+                    properties.remove("name").toString();
+            String path = !(properties.get("path") instanceof String) ? null :
+                    properties.remove("path").toString();
+
+            if (name != null && !name.isEmpty() && path != null) {
+                if (!(new File(path).exists())) {
+                    try {
+                        URL url = new URL(path);
+                        path = loadRemoteJAR(targetDir, name, url, outputStream);
+                    } catch (MalformedURLException ex) {
+                        throw new RuntimeException("Failed to retrieve remote platform file with exception: "
+                                + ex.getMessage());
+                    }
+                }
+
+                if (Files.exists(Path.of(path))) {
+                    ScanTomlFile.Platform platform = new ScanTomlFile.Platform(name, path, properties);
+                    scanTomlFile.setPlatform(platform);
+                }
+            }
+        });
+
+        List<Toml> staticCodeAnalyzerTables = scanTomlDocumentContent.getTables(ANALYZER_TABLE);
+        staticCodeAnalyzerTables.forEach(staticCodeAnalyzer -> {
+            Map<String, Object> properties = staticCodeAnalyzer.toMap();
+            String org = !(properties.get("org") instanceof String) ? null :
+                    properties.get("org").toString();
+            String name = !(properties.get("name") instanceof String) ? null :
+                    properties.get("name").toString();
+
+            Pattern versionPattern = Pattern.compile(CUSTOM_RULES_COMPILER_PLUGIN_VERSION_PATTERN);
+            Object providedVersion = properties.get("version");
+            String version = (providedVersion instanceof String) &&
+                    versionPattern.matcher(providedVersion.toString()).matches() ? providedVersion.toString() : null;
+            String repository = !(properties.get("repository") instanceof String) ? null :
+                    properties.get("repository").toString();
+
+            if (org != null && !org.isEmpty() && name != null && !name.isEmpty()) {
+                ScanTomlFile.Analyzer analyzer;
+
+                if (repository != null && repository.equals(LOCAL_REPOSITORY_NAME) && version != null &&
+                        !version.isEmpty()) {
+                    analyzer = new ScanTomlFile.Analyzer(org, name, version, repository);
+                } else {
+                    analyzer = new ScanTomlFile.Analyzer(org, name, version, null);
+                }
+                scanTomlFile.setAnalyzer(analyzer);
+            }
+        });
+
+        Toml rulesTable = scanTomlDocumentContent.getTable(RULES_TABLE).orElse(null);
+        if (rulesTable != null) {
+            TomlValueNode include = rulesTable.get("include").orElse(null);
+            TomlArrayValueNode includeArray;
+            if (include != null && include.kind().equals(TomlType.ARRAY)) {
```

Comment:
IMO it is cleaner to chain `filter`, `ifPresent` here
```suggestion
            rulesTable.get("include").filter(include -> include.kind().equals(TomlType.ARRAY)).ifPresent(include -> {
            ....
```